kod/object/active/holder/room/battlearena.kod

```diff
@@ -0,0 +1,1095 @@
+      
+      oHoldSpell = Send(SYS,@FindSpellByNum,#num=SID_HOLD);
+      Send(oHoldSpell,@DoHold,#otarget=victim,
+                              #idurationsecs=viDeathHoldSeconds,
+                              #report=FALSE,
+                              #bAllowFreeAction=FALSE);
+
+      for i in Send(self,@GetHolderActive)
+      {
+         each_object = Send(self,@HolderExtractObject,#data=i);
+         if IsClass(each_object,&User)
+         {
+            Send(each_object,@MsgSendUser,#message_rsc=kill_reward_jackpot,
+                                          #parm1=Send(what,@GetTrueName),
+                                          #parm2=Send(victim,@GetTrueName),
+                                          #parm3=piJackpot);
+         }
+      }
+      
+      propagate;
+   }
+   
+   IncreaseJackpot(amount=0)
+   "Admin supported."
+   {
+      piJackpot = piJackpot + amount;
+      
+      if piJackpot > JACKPOT_MAX
+      {
+         piJackpot = JACKPOT_MAX;
+      }
+      return;
+   }
+
+   OpenDoorDown()
+   {
+      return;
+   }
+
+   OpenSecondaryDoorDown()
+   {
+      Send(self,@SetSector,#sector=SECTOR_DOOR_CRYPT3A,
+           #animation=ANIMATE_CEILING_LIFT,#height=1098,#speed=8);
+       
+      return;
+   }
+
+   CloseSecondaryDoorDown()
+   {
+      Send(self,@SetSector,#sector=SECTOR_DOOR_CRYPT3A,
+           #animation=ANIMATE_CEILING_LIFT,#height=1098,#speed=8);
+              
+      return;
+   }
+   
+   ReqSomethingAttack(what = $,victim = $,use_weapon = $)
+   "Opposite teams may attack each other. Spectators may not interfere."
+   {
+      local iAttackerTeam, iVictimTeam;
+   
+      if IsClass(what, &User)
+      {
+         iAttackerTeam = Send(self,@GetUserTeam,#what=what);
+         iVictimTeam = Send(self,@GetUserTeam,#what=victim);
+      
+         if iAttackerTeam = TEAM_SPECTATOR
+         {
+            Send(what,@MsgSendUser,#message_rsc=no_participation_spectators);
+            return FALSE;
+         }
+
+         if iAttackerTeam = iVictimTeam
+         {
+            Send(what,@MsgSendUser,#message_rsc=no_attack_teammate);
+            return FALSE;
+         }
+         
+         if iAttackerTeam <> iVictimTeam
+            AND iVictimTeam <> TEAM_SPECTATOR
+         {
+            return TRUE;
+         }
+      }
+
+      return TRUE;
+   }
+
+   ReqSpellCast(who = $, oSpell = $, lItems = $)
+   "Spectators may not cast here."
+   {
+      if Send(self,@GetUserTeam,#what=who) = TEAM_SPECTATOR
+      {
+         Send(who,@MsgSendUser,#message_rsc=no_participation_spectators);
+         return FALSE;
+      }
+
+      if oSpell = Send(SYS,@FindSpellByNum,#num=SID_SWEEP)
+         OR oSpell = Send(SYS,@FindSpellByNum,#num=SID_SHATTER)
+         OR oSpell = Send(SYS,@FindSpellByNum,#num=SID_SWAP)
+         OR oSpell = Send(SYS,@FindSpellByNum,#num=SID_BRITTLE)
+         OR oSpell = Send(SYS,@FindSpellByNum,#num=SID_TRUCE)
+         OR oSpell = Send(SYS,@FindSpellByNum,#num=SID_JIG)
```

Comment:
Instead of this large check, put a new message in spell.kod CanCastInBattleArena() { return TRUE; } then set it to FALSE for the spells you don't want cast there. Could also make it a property of the spell if you wanted.